### PR TITLE
[WiP] Using API improvements around Executor/Executable

### PR DESCRIPTION
**Pull request by jglick**
Based on https://github.com/jenkinsci/jenkins/pull/1610. Subsumes #94 and #96.

@reviewbybees (WiP until 1.607 is in an LTS line)

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Should be squashed into a single commit before integrating in #120

File: job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java [hunk 1]
```diff
@@ -194,39 +195,53 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
...
             executionPromise.setException(x);
+            return;
         }
+        throw sleep();
     }
 
-    /**
-     * Sleeps until the run is finished, updating log messages periodically.
-     */
-    void waitForCompletion() {
```
Comment (oleg-nenashev) [inline job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java]: Is it safe? the method was not private

**Comment by jglick** [comment]
I do not rebase. `git-merge` works fine.

File: job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java (hunk 1 above)
Comment (jglick) [inline job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java]: Package-private, and anyway the `workflow-job` plugin was not intended to expose any APIs.

File: support/src/main/resources/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution/PlaceholderTask/PlaceholderExecutable/executorCell.jelly [hunk 2]
```diff
@@ -26,10 +26,23 @@
 
 <j:jelly xmlns:j="jelly:core" xmlns:t="/lib/hudson" xmlns:l="/lib/layout">
     <td class="pane">
-        <div style="white-space: normal">${%Building}
-            <!-- TODO check for Item.READ permission on the accessControlled() -->
-            <a href="${rootURL}/${it.parent.url}" class="model-link"><l:breakable value="${it.parent.fullDisplayName}"/></a>
-            <t:buildProgressBar build="${it.parent}" executor="${executor}"/>
+        <div style="white-space: normal">
+            <j:set var="r" value="${it.parent.run()}"/>
+            <j:set var="rAuth" value="${r != null and h.hasPermission(r.parent, r.parent.READ) ? r : null}"/>
```
Comment (oleg-nenashev) [inline support/src/main/resources/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution/PlaceholderTask/PlaceholderExecutable/executorCell.jelly]: <code>Item.DISCOVER</code> may be missing in the logic (e.g. "I can see the project, but cannot open its internals"). On the other hand, this executor cell will leak some info in such case

**Comment by oleg-nenashev** [comment]
LGTM

File: support/src/main/resources/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution/PlaceholderTask/PlaceholderExecutable/executorCell.jelly (hunk 2 above)
Comment (jglick) [inline support/src/main/resources/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution/PlaceholderTask/PlaceholderExecutable/executorCell.jelly]: At any rate, this logic is basically copied from freestyle projects.

**Comment by jglick** [comment]
Merged via #120 into `lts-609`.

**Pull request closed without merging**